### DISCUSSION TITLE
2FA problem

### DISCUSSION THREAD

**Discussion by muratdincer** [PiotrMachowski/Xiaomi-cloud-tokens-extractor#92] [Q&A] [answered]
Hello!

I’ve run extractor exe app in my windows machine. But later I’ve realized 2FA link should be open in home assistant app machine. I tried in ssh terminal but every time it’s getting same token before I got from windows machine. Token extractor says failed to login. Can you help me please.

**Comment by PiotrMachowski** [comment]
Tokens are the same regardless the device that was used to retrieve them. Where do you get "failed to login"?

**Reply by done7k** [reply]
I have been experiencing an issue to connect to Xiaomi application server from Mi Home: 10apr and 11apr definitely, I could not use my vacuum due to this issue. (Hate this architecture to have maps on the server.. and each mobile app take 1Gb of memory and anyway require the maps on the server...)  Maybe this somehow relates to the issue above?

**Reply by muratdincer** [reply]
Hi


<img width="1180" alt="image" src="https://user-images.githubusercontent.com/2674696/231491131-1dc8ebbf-2f28-4ac8-a4a3-86be3dd59039.png">

**Reply by PiotrMachowski** [reply, marked as answer]
@muratdincer https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/issues/422

**Comment by zpavlovic1986** [comment]
fixed xiaomi cloud token extractor with 2FA: https://github.com/zpavlovic1986/xiaomi_token_extractor_2025/tree/main
